src/ol/expr/expression.js

```diff
@@ -11,6 +11,110 @@ import {fromString, isStringColor} from '../color.js';
  * expression.
  */
 
+/**
+ * Base type used for literal style parameters; can be a number literal or the output of an operator,
+ * which in turns takes {@link import("./expression.js").ExpressionValue} arguments.
+ *
+ * The following operators can be used:
+ *
+ * * Reading operators:
+ *   * `['band', bandIndex, xOffset, yOffset]` For tile layers only. Fetches pixel values from band
+ *     `bandIndex` of the source's data. The first `bandIndex` of the source data is `1`. Fetched values
+ *     are in the 0..1 range. {@link import("../source/TileImage.js").default} sources have 4 bands: red,
+ *     green, blue and alpha. {@link import("../source/DataTile.js").default} sources can have any number
+ *     of bands, depending on the underlying data source and
+ *     {@link import("../source/GeoTIFF.js").Options configuration}. `xOffset` and `yOffset` are optional
+ *     and allow specifying pixel offsets for x and y. This is used for sampling data from neighboring pixels.
+ *   * `['get', 'attributeName', typeHint]` fetches a feature property value, similar to `feature.get('attributeName')`
+ *     A type hint can optionally be specified, in case the resulting expression contains a type ambiguity which
+ *     will make it invalid. Type hints can be one of: 'string', 'color', 'number', 'boolean', 'number[]'
```

Comment:
Doesn't have to be here, but I think it would be good to move away from these type hints in favor of type assertions (like those [in the Mapbox style spec](https://docs.mapbox.com/style-spec/reference/expressions/#type-system)).

For example:
```json
["<", ["number", ["get", "a"]], ["number", ["get", "b"]]]
```

Reply:
Assertions might be good indeed, and I think we already don't need them in 99% of the cases. In your example the number type would already be inferred from the operator.

On the other hand I liked the typehint on `get` because it is very unintrusive and `get` is really the only operator where it might be impossible to infer a single type.